MDANSE_GUI/Src/MDANSE_GUI/Tabs/Visualisers/Action.py

```diff
@@ -236,18 +237,21 @@ def update_panel(self, job_name: str) -> None:
             dtype = value[0]
             ddict = value[1]
             configurator = job_instance.configuration[key]
-            ddict.setdefault("label", key)
-            ddict["configurator"] = configurator
-            ddict["source_object"] = self._input_trajectory
-            widget_class = widget_lookup[dtype]
-            input_widget = widget_class(parent=self, **ddict)
-            widget = input_widget._base
-            self.layout.addWidget(widget, stretch=input_widget._relative_size)
-            self._widgets_in_layout.append(widget)
-            self._widgets.append(input_widget)
-            self._trajectory_configurator = input_widget._configurator
+            if key not in self._widgets_in_layout:
+                ddict.setdefault("label", key)
+                ddict["configurator"] = configurator
+                ddict["source_object"] = self._input_trajectory
+                widget_class = widget_lookup[dtype]
+                input_widget = widget_class(parent=self, **ddict)
+                widget = input_widget._base
+                self.layout.addWidget(widget, stretch=input_widget._relative_size)
+                self._widgets_in_layout[key] = widget
+                self._widgets.append(input_widget)
+                self._trajectory_configurator = input_widget._configurator
             LOG.info("Set up input trajectory")
         for key, value in settings.items():
+            if key in self._widgets_in_layout:
+                continue
             if key == "trajectory":
                 continue
```

Comment:
```suggestion
            if key in self._widgets_in_layout or key == "trajectory":
                continue
```
Or is `trajectory` in `widgets_in_layout` already?

Reply:
You are right, "trajectory" should be pretty much guaranteed either to be already in the `_widgets_in_layout`, or not to appear at all.